plugins/org.yakindu.sct.ui.editor/src/org/yakindu/sct/ui/editor/definitionsection/StatechartDefinitionSection.java

```diff
@@ -766,4 +783,30 @@ public void dispose() {
 			super.dispose();
 		}
 	}
+
+	/**
+	 * @author BeckmaR
+	 *
+	 *         Adapter for updating the name label on statechart name change
+	 *         event. Suppresses the nameLabel's ModificationEventListener to
+	 *         avoid a StackOverflowException. Shamelessly copied from
+	 *         {@link org.yakindu.sct.ui.editor.partitioning.DiagramPartitioningEditor.BreadcrumbSynchronizer}
+	 */
+	protected class DefinitionSectionSynchronizer extends AdapterImpl {
+		@Override
+		public void notifyChanged(Notification notification) {
+			if (Notification.SET == notification.getEventType()) {
+				Object feature = notification.getFeature();
+				if (BasePackage.Literals.NAMED_ELEMENT__NAME.equals(feature)) {
+					suppressModifyEvent = true;
+					nameLabel.setText(notification.getNewStringValue());
+					suppressModifyEvent = false;
```

Comment:
Wouldn't it be enough to just check if new name and old name are different, and only set if both are not equal, to avoid endless notification cycles?

Reply:
Well, this would still trigger the whole notification cycle twice, which might have some implications now or in the future.